int8        nOperations
string[]    pickPoses
string[]    placePoses